templates/SociIndexBuilder.yml

```diff
@@ -43,6 +43,15 @@ Parameters:
       the template. Changing the prefix updates code references to point to 
       a new location.
     Type: String
+  IamPermissionsBoundaryArn:
+    Type: String
+    Description: >-
+      IAM Roles might require an IAM Permissions boundary in order to be created and
+      perform subsequent API calls to services, IAM itself included.
+      This parameter expects the ARN of an IAM policy, or to be set to none.
+      If set to None, the PermissionsBoundary property is omitted on IAM Role creation.
+    Default: none
+    AllowedPattern: none|^arn:aws(?:):iam::[\d]{12}:policy/[0-9a-zA-Z!-_\.\*'\(\)/]+$
```

Comment:
Does this regex work for other partitions? It looks like you added a non-capturing group for it, but nothing is allowed to go there, right?

Reply:
`aws(?:-cn|-us-gov)?` would explicitly handle the two other partitions, or a more generic `aws(?:-.+)?` would handle hypothetical future partitions.

Reply:
Sorry, I already approved this, but I think it would be better to use a more generic pattern. It's similar to @sparr's but slightly more restrictive:
```
^arn:aws(?:-[^:]+)?:iam::[\d]{12}:policy/[0-9a-zA-Z!-_\.\*'\(\)/]+$
```
https://regex101.com/r/wINLY5/1